OpenRA.Game/Traits/Player/FrozenActorLayer.cs

```diff
@@ -80,7 +80,7 @@ public FrozenActor(Actor self, PPos[] footprint, Shroud shroud, bool startsRevea
 			Bounds = self.Bounds;
 			TargetTypes = self.GetEnabledTargetTypes().ToHashSet();
 
-			tooltip = self.TraitsImplementing<ITooltip>().FirstOrDefault();
+			tooltip = self.TraitsImplementing<ITooltip>().FirstOrDefault(Exts.IsTraitEnabled);
```

Comment:
You'll need to store all of these here, and then pick the appropriate one in `RefreshState`.